src/internal/packager/helm/chart.go

```diff
@@ -58,28 +61,35 @@ func (h *Helm) InstallOrUpgradeChart(ctx context.Context) (types.ConnectStrings,
 	}
 
 	histClient := action.NewHistory(h.actionConfig)
+	var release *release.Release
+
+	var helmCtx context.Context
+	var helmCtxCancel context.CancelFunc
 
 	err = retry.Do(func() error {
 		var err error
 
 		releases, histErr := histClient.Run(h.chart.ReleaseName)
 
 		spinner.Updatef("Checking for existing helm deployment")
-
+		// cancel context from previous retry if it exists
+		if helmCtxCancel != nil {
+			helmCtxCancel()
+		}
+		helmCtx, helmCtxCancel = context.WithTimeout(ctx, h.timeout)
```

Comment:
I think everything would be simpler if we did not reset the timeout every time. Is there a reason we  do not create a single context with timeout in the beginning of the function call?

Reply:
I did not originally do this because it affects the timeout on retries. Given how successful retries seemingly never happen in this situation however, I agree it's better to go the simpler route